### DISCUSSION TITLE
IndexError: index 42 is out of bounds for axis 0 with size 40

### DISCUSSION THREAD

**Discussion by celestinoxp** [pycaret/pycaret#2351] [General]
Hi,

I have some questions about if pycaret have a bug or its mycode not correct working...

after this code: 
best_model = compare_models()

I print config with this code: print(best_model)
and pycaret return this:
ExtraTreesClassifier(bootstrap=False, ccp_alpha=0.0, class_weight=None,
                     criterion='gini', max_depth=None, max_features='auto',
                     max_leaf_nodes=None, max_samples=None,
                     min_impurity_decrease=0.0, min_impurity_split=None,
                     min_samples_leaf=1, min_samples_split=2,
                     min_weight_fraction_leaf=0.0, n_estimators=100, n_jobs=1,
                     oob_score=False, random_state=1345, verbose=0,
                     warm_start=False)

My problem its after this: tuned_data = tune_model(best_model, choose_better = True)
pycaret return error:
IndexError: index 42 is out of bounds for axis 0 with size 40

My question: it is not possible tune this algorithm?

**Comment by ngupta23** [comment]
It is not possible to provide an answer with the information you provided. Can you provide a reproducible example (complete code + dataset)?
